### PR TITLE
doc/architecture: "Edit HA Auth"

### PR DESCRIPTION
**Pull request by zdover23**
Rewrite the explanation of how a client authenticates against a monitor. This is a rewrite of a single paragraph, and has been set apart in its own PR so that it can receive the maximum amount of scrutiny that the upstream Ceph community can muster.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/architecture.rst [hunk 1]
```diff
@@ -278,16 +278,17 @@ the client and the monitor share a secret key.
...
-monitor, and the monitor generates a session key and encrypts it with the secret
-key associated to the user name. Then, the monitor transmits the encrypted
-ticket back to the client. The client then decrypts the payload with the shared
-secret key to retrieve the session key. The session key identifies the user for
-the current session. The client then requests a ticket on behalf of the user
-signed by the session key. The monitor generates a ticket, encrypts it with the
-user's secret key and transmits it back to the client. The client decrypts the
-ticket and uses it to sign requests to OSDs and metadata servers throughout the
-cluster.
+Here are the steps in the process whereby a client authenticates with a
```
Comment (anthonyeleven) [inline doc/architecture.rst:281]: Seems a bit stilted.

Here are the steps when a client authenticates with a monitor.

File: doc/architecture.rst [hunk 2]
```diff
@@ -278,16 +278,17 @@ the client and the monitor share a secret key.
...
+monitor. The client passes the user name to the monitor. The monitor generates
```
Comment (anthonyeleven) [inline doc/architecture.rst:282]: suggest `username`

File: doc/architecture.rst [hunk 3]
```diff
@@ -278,16 +278,17 @@ the client and the monitor share a secret key.
...
+a session key that is encrypted with the secret key associated with the user
+name. The monitor transmits the encrypted ticket to the client. The client
+uses the shared secret key to decrypt the payload, which is the session key.
```
Comment (anthonyeleven) [inline doc/architecture.rst:285]: End after "payload", we already described the payload.
maybe also s/payload/session key/

**Review by anthonyeleven — APPROVED** [review]

File: doc/architecture.rst (hunk 1 above)
Comment (zdover23) [reply doc/architecture.rst:281]: I don't like the adverb of time "when", so I'm going with the adverb of manner "how":

> Here is how a client authenticates with a monitor.

Maybe that will make it sound a bit less like it was written by Philboyd Studge.

File: doc/architecture.rst (hunk 2 above)
Comment (zdover23) [reply doc/architecture.rst:282]: Accepted.

File: doc/architecture.rst (hunk 3 above)
Comment (zdover23) [reply doc/architecture.rst:285]: Yeah, okay. I think you're right. That is better.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53619 - Reef backport
https://github.com/ceph/ceph/pull/53620 - Quincy backport

**Pull request merged**